processor/patch.go

```diff
@@ -30,10 +34,22 @@ func patchFile(fset *token.FileSet, file *ast.File, patches ...patch) error {
 	for _, patch := range patches {
 		buf.Reset()
 
+		// Insert the instrumentation statements
 		buf.WriteRune('\n')
 		if err := format.Node(&buf, fset, patch.stmts); err != nil {
 			return err
 		}
+		
+		// Add line directive to preserve original line numbers for the first original statement
```

Comment:
add documentation how this is achieved. 
```suggestion
		// line directive to preserve original line numbers for the first original statement
		// https://github.com/golang/go/blob/master/src/cmd/compile/doc.go#L171
```